### ISSUE TITLE
Steam shortcut error - double click on the main exe OK

### ISSUE THREAD

**Issue by donizettilorenzo** [narzoul/DDrawCompat#68]
Lastest binaries of ddrawcompat
Unreal Gold from Steam (NO other patches)

If I launch it through the shortcut/Steam library "Play" button -> CRASH with no particolar log:
_Warning: glxStopOutput() failed: 6
Exit: Executing UWindowsClient::ShutdownAfterError
Exit: UD3DRenderDevice::ShutdownAfterError
Exit: appExit
Uninitialized: Name subsystem shut down
Uninitialized: Log file closed, 05/21/20 14:46:52_

If I launch it through the main exe (Steam being active) Unreal.exe -> Allright

**Comment by narzoul** [comment]
What does the log of DDrawCompat say? Does the game work without it?

**Comment by donizettilorenzo** [comment]
The game works without it (but in the expected classic 60 FPS locked manner)
No particular information in DDrawCompat log!

**Comment by donizettilorenzo** [comment]
Here's the log

_1f04 16:04:43.673 Process path: E:\Steam Games\steamapps\common\Unreal Gold\system\Unreal.exe
1f04 16:04:43.673 Environment variable __COMPAT_LAYER = ""
1f04 16:04:43.673 Loading DDrawCompat dynamically from E:\Steam Games\steamapps\common\Unreal Gold\system\ddraw.dll
1f04 16:04:43.677 DDrawCompat loaded successfully
1f04 16:04:43.677 Installing display mode hooks
1f04 16:04:43.684 Installing registry hooks
1f04 16:04:43.684 Installing Direct3D driver hooks
1f04 16:04:43.685 Installing Win32 hooks
1f04 16:04:43.773 Hooking user mode display driver: C:\WINDOWS\System32\DriverStore\FileRepository\iigd_dch.inf_amd64_cbd42a415845f91a\igdumdim32.dll
1f04 16:04:43.805 Installing DirectDraw hooks
1f04 16:04:43.808 Installing Direct3D hooks
1f04 16:04:43.826 Installing GDI hooks
1f04 16:04:43.947 Finished installing hooks
1f04 16:04:48.092 Detaching DDrawCompat due to process termination
1f04 16:04:48.093 DDrawCompat detached successfully_

**Comment by narzoul** [comment]
Indeed, nothing suspicious there. Please upload the debug logs produced by the ReleaseWithDebugLogs build (in ddraw-debug.zip).

**Comment by donizettilorenzo** [comment]
With "release with debug logs" the rendering is passed to the software renderer......(launching with the Steam Shortcut).
Here's the log: 
[DDrawCompat-Unreal.zip](https://github.com/narzoul/DDrawCompat/files/4663741/DDrawCompat-Unreal.zip)

**Comment by narzoul** [comment]
Thanks! I don't have the Steam version of the game, but I have the GOG version, and I managed to reproduce the problem by importing that to my Steam library and running the game through the Steam client. Now I have the same issue as you, it fails with 80070057 (E_INVALIDARG) when trying to change display mode (IDirectDrawVtbl::SetDisplayMode in the logs).

I think it must be some kind of interference from GameOverlayRenderer.dll that Steam injects for its overlay to work. Unfortunately, it seems that even disabling the overlay from settings will not disable this injection. I actually found that it writes some logs too in GameOverlayRenderer.log, next to the dll file (in C:\Program Files (x86)\Steam by default), which shows that it hooks DirectDraw also.

With multiple hooks from different sources, there can always be conflicts like this. There are similar issues reported already for some video capture tools (see issue #13). I'll try to fix these if I can, but it's currently not at the top of my priority list, so it might take a while longer...
